app/views/assessments/viewFeedback.html.erb

```diff
@@ -1,24 +1,103 @@
 <% if @jsonFeedback.nil? %>
 
-<h2>Feedback for
+<h2>Autograding Result for
 <%= "#{@assessment.display_name} - #{@score.problem.name}" %>
 (<%=@score.submission.course_user_datum.user.email %>)
 </h2>
-<pre>
-<%= @score.feedback %>
-Score for this problem: <%= @score.score.to_f().round(1) %>
+<div class="result-container">
+  <div class="feedback-container">
+    <div class="result-pane">
+      <div class="feedback-header">
+        <h3>Feedback</h3>
+        <div class="feedback-status__completed">
+          Completed
+        </div>
+      </div>
+      <pre><%= @score.feedback %></pre>
+    </div>
+    <% if @submission.annotations.count > 0 and @problemReleased %>
+      <h2 style="margin-top: 1em">Remarks</h2>
 
-Graded by: <%=
-  case @score.grader_id
-  when -1
-    "Error Handler"
-  when 0
-    "Autograder"
-  else
-    if @score.grader&.user then @score.grader.user.email end
-  end
-%>
-</pre>
+      <% subs_by_problem = @submission.annotations.group_by(&:problem)
+          subs_by_problem.each do |problem, annotations| %>
+          <b> <%= problem.name %> </b>
+          <ul class="collection">
+          <!-- use custom sort b/c using ordering doesn't work with autograder output, archived files -->
+          <%
+            s_annotations = annotations.sort do |a,b|
+            aFilename = @get_correct_filename.call(a)
+            bFilename = @get_correct_filename.call(b)
+            case
+              when aFilename < bFilename
+                -1
+              when aFilename > bFilename
+                  1
+              else
+                  a.line <=> b.line
+              end
+            end
```

Comment:
@damianhxy was there a helper function for this?

Reply:
Idts, #1590 introduced this code, which only made a helper function `get_correct_filename` and not one for sorting annotations